python/semantic_kernel/contents/chat_history.py

```diff
@@ -97,7 +98,7 @@ def add_message(
             encoding (Optional[str]): The encoding of the message. Required if 'message' is a dict.
             metadata (Optional[dict[str, Any]]): Any metadata to attach to the message. Required if 'message' is a dict.
         """
-        if isinstance(message, ChatMessageContent):
+        if isinstance(message, ChatMessageContent) or isinstance(message, StreamingChatMessageContent):
```

Comment:
maybe we should cast the combined StreamingCMC back into a CMC since they do not stem from the same source so some type checking might fail when encountering a StreamingCMC instead of CMC...

Reply:
I don't know if I follow. Could you elaborate, please? What type checking would fail?